Automattic/jetpack
# Publicize: Create new objects in the store effect

Previously we would copy over the any existing object based on the
connection ID, which meant that if any data had been updated (like the
profile image) it wouldn't be seen as a change, because the object
reference would still be the same.

This creates a new object for each connection in response to the API
results. It will force a rerender, but I think that would happen anyway
as we dispatch the action to update the meta data already.

We could potentially check for any of the properties having changed
instead. I'm not sure if this is a necessary optimisation though.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

This came up in testing of D69728-code. One way to test this is to:

* Sandbox the API
* Follow the instructions in D69728-code to update the Facebook profile image, so that it's broken. 
* Load the editor and open the Jetpack sidebar
* Apply D69728-code to your sandbox
* Switch tab, and switch back, or close and open the Jetpack sidebar to trigger a refresh of the connections
* Notice that the API response has an updated profile image, but that it's not rendered.
* Check out this branch and repeat the process. You'll need to remove D69728-code from your sandbox to ensure that the profile image isn't fixed on the initial load.
